File: IterativePreorderTraversal.cpp
#include<iostream>
using namespace std;

class Solution{
    public:
    vector<int> preOrderTraversal(TreeNode* root){
        vector<int> preorder;

        //Base Case:
        if(root == NULL)
           return preorder;
        
        stack<TreeNode*> st;
        st.psuh(root);
        while(!st.empty()){
            root= st.top();
            st.pop();

            preorder.push_back(root->val);

            if(root->right != NULL){
                st.push(root->right);
            }

            if(root->left!= NULL){
                st.push(root->left);
            }

        }
        return preorder;
    }
};

File: IterativePreorderTraversal.java
import java.util.*;
import java.io.*;

class Solution{
    public List<Integer> preorderTraversal(TreeNode root){
        List<Integer> preorder = new ArrayList<Integer>();

        //Base Case: 
        if(root == null)
            return preorder;
        
        Stack<TreeNode> st = new Stack<TreeNode>();
        st.psuh(root);
        while(!st.isEmpty()){
            root=st.top();
            st.pop();

            preorder.add(root.value);

            if(root.right != null){
                st.push(root.right);
            }

            if(root.left != null){
                st.push(root.left);
            }
        }
        return preorder;
    }
}